### PR TITLE
Use GitHub Actions as CI service

### PR DESCRIPTION
**Pull request by liushuyu**
As previously discussed, Travis CI is going to use a new pricing model which is unaffordable for Citra.

So this Pull Request will migrate the CI service to GitHub Actions. 

In addition, other adaptive changes are made:

- Windows MSVC build is added (in attempt to replace Appveyor)
- To cache MSVC build better, debug info emission is disabled if `USE_CCACHE` variable is set to true
- The `scan_dll` script now uses `lief` library for parsing PE files (significant speed up comparing to `pefile`)

Caveats:

- Due to an issue with one of the GitHub Actions library, during release, several errors will be displayed. However, this will not fail or affect the release process.
- **Set up is required prior to merging this Pull Request.** Please go to "Settings" -> "Secrets" and add a secret with the name `TRANSIFEX_API_TOKEN` and the value set to the Transifex API token found in the Transifex User Settings Panel ("Profile" -> "User Settings" -> "API Tokens")

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5602)

### CODE REVIEW

File: .ci/clang-format/docker.sh [hunk 1]
```diff
@@ -1,5 +1,4 @@
 #!/bin/bash -ex
 
 # Run clang-format
-cd /citra
 ./.travis/clang-format/script.sh
```
Comment (B3n30) [inline .ci/clang-format/docker.sh:4]: why is this line not changed?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        run: ./.ci/${{ matrix.image }}/upload.sh
+        if: ${{ matrix.image != 'linux-frozen' }}
+      - name: Upload
+        uses: actions/upload-artifact@v2
+        if: ${{ matrix.image != 'linux-frozen' }}
+        with:
+          name: ${{ matrix.image }}
+          path: artifacts/
+  macos:
+    runs-on: macos-latest
```
Comment (B3n30) [inline .github/workflows/ci.yml:43]: should we use an older version here (if possible)?

**Review by B3n30 — APPROVED** [review]
Thx for addressing this

From a first glance it looks fine to me

File: .ci/clang-format/docker.sh (hunk 1 above)
Comment (liushuyu) [reply .ci/clang-format/docker.sh:4]: Sorry, I made a mistake. I will fix it in the next commit.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (liushuyu) [reply .github/workflows/ci.yml:43]: The VM used here currently is macOS 10.15 (Catalina). Unfortunately, they only have 10.15 and 11.0 (Big Sur).

File: .ci/clang-format/docker.sh (hunk 1 above)
Comment (liushuyu) [reply .ci/clang-format/docker.sh]: Should be fixed now

**Review by zhaowenlan1779 — APPROVED** [review]

**Pull request merged**